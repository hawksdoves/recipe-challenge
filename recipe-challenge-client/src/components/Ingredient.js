import React from 'react';

function Ingredient(props) {
    const { ingredient } = props

    return (
        <div key={ingredient.name}>
            { ingredient.quantity ?
                ingredient.measure ? 
                    `${ingredient.quantity} ${ingredient.measure} ` :
                    `${ingredient.quantity} x ` :
                    null
            }
            {ingredient.name}
        </div>
    )
};

export default Ingredient;