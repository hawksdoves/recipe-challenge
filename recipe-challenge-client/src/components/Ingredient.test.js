import React from 'react';
import { shallow } from 'enzyme';

import Ingredient from './Ingredient';

describe('Ingredient', () => {
    it('should render the ingredient and measurement correctly ', () =>{
        const ingredient = {
            name: 'thyme',
            quantity: '3',
            measure: 'tsp'
        }

        const wrapper = shallow(<Ingredient ingredient={ingredient} />);

        expect(wrapper).toMatchSnapshot()
    })

    it('should render only the ingredient and quantity correctly ', () =>{
        const ingredient = {
            name: 'steak',
            quantity: '3'
        }

        const wrapper = shallow(<Ingredient ingredient={ingredient} />);

        expect(wrapper).toMatchSnapshot()
    })

    it('should render only the ingredient correctly ', () =>{
        const ingredient = {
            name: 'steak'
        }

        const wrapper = shallow(<Ingredient ingredient={ingredient} />);

        expect(wrapper).toMatchSnapshot()
    })
});
