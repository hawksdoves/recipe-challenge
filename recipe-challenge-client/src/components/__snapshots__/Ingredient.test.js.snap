// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ingredient should render only the ingredient and quantity correctly  1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    3 x 
    steak
</div>,
  "nodes": Array [
    <div>
      3 x 
      steak
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Ingredient
        ingredient={
                Object {
                        "name": "steak",
                        "quantity": "3",
                      }
        }
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "ingredient": Object {
            "name": "steak",
            "quantity": "3",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          3 x 
          steak
</div>,
        "_debugID": 4,
        "_renderedOutput": <div>
          3 x 
          steak
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Ingredient
    ingredient={
        Object {
            "name": "steak",
            "quantity": "3",
          }
    }
/>,
}
`;

exports[`Ingredient should render only the ingredient correctly  1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    steak
</div>,
  "nodes": Array [
    <div>
      steak
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Ingredient
        ingredient={
                Object {
                        "name": "steak",
                      }
        }
/>,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "ingredient": Object {
            "name": "steak",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          steak
</div>,
        "_debugID": 6,
        "_renderedOutput": <div>
          steak
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Ingredient
    ingredient={
        Object {
            "name": "steak",
          }
    }
/>,
}
`;

exports[`Ingredient should render the ingredient and measurement correctly  1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    3 tsp 
    thyme
</div>,
  "nodes": Array [
    <div>
      3 tsp 
      thyme
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Ingredient
        ingredient={
                Object {
                        "measure": "tsp",
                        "name": "thyme",
                        "quantity": "3",
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "ingredient": Object {
            "measure": "tsp",
            "name": "thyme",
            "quantity": "3",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          3 tsp 
          thyme
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          3 tsp 
          thyme
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Ingredient
    ingredient={
        Object {
            "measure": "tsp",
            "name": "thyme",
            "quantity": "3",
          }
    }
/>,
}
`;
